src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/constructorsdeclarationgrouping/InputConstructorsDeclarationGroupingRecords.java

```diff
@@ -0,0 +1,76 @@
+/*
+ConstructorsDeclarationGrouping
+
+
+*/
+
+//non-compiled with javac: Compilable with Java14
+package com.puppycrawl.tools.checkstyle.checks.coding.constructorsdeclarationgrouping;
+
+public class InputConstructorsDeclarationGroupingRecords {
+    public record MyRecord1(int x, int y) {
+        public MyRecord1(int a) {
+            this(a,a);
+        }
+
+        void foo() {}
+
+        void foo2() {}
+
+        public MyRecord1 {} // violation
+
+        public MyRecord1(int a, int b, int c, int d) { // violation
+            this(a+b, c+d);
+        }
+
+        public MyRecord1(int x, int y, int z) { // violation
+            this(x+y, z);
+        }
+    }
+
+    class MyClass {
+        int x = 20;
+
+        MyClass() {}
+
+        MyClass(String s) {}
+
+        String[] str;
+
+        String[] str2;
+
+        MyClass(int a) {} // violation
```

Comment:
Please add violation message to all violations

Reply:
you mean something like this right?

https://github.com/checkstyle/checkstyle/blob/e7facb06e1bf121b8715fa82f406a24ab5bf6c0b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/sizes/methodlength/InputMethodLengthCountEmptySmallSize.java#L45

I tried adding it with using RegEx to make the message shorter but it is still crossing the 100+ column limit. For Example:


```
InputConstructorsDeclarationGrouping(int a) {} // violation 'Constructors should be .* together.*'
```

This line is of 103 columns even with using RegEx. 

Do we need to compulsory add it? I mean the check has only one violation so I guess it is okay if we don't specify the violation message as there will be only one fixed violation will be logged when using this check.

Reply:
You can use violation above or below 

> So we don't need to explicitly mention that which violation will be printed

yes, IMO

Reply:
@nrmancuso I have added the violation message as you suggested: https://github.com/checkstyle/checkstyle/blob/b1bbce36e3704d0a1af1f9ff0e57f9a0721d40ae/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/constructorsdeclarationgrouping/InputConstructorsDeclarationGrouping.java#L23-L24

I have used RegEx to make the message shorter, but can you pls take a look at this comment also?https://github.com/checkstyle/checkstyle/pull/14844#discussion_r1598887608

Reply:
@nrmancuso